server/src/main/java/io/littlehorse/common/model/getable/core/usertaskrun/UserTaskRunModel.java

```diff
@@ -297,6 +299,17 @@ public void cancel() {
         failureToThrowKenobi = new FailureModel("User task cancelled", failureName);
     }
 
+    public void processProgressSavedEvent(SaveUserTaskRunProgressRequestModel req, ProcessorExecutionContext ctx)
+            throws LHApiException {
+        if (isTerminated()) {
+            throw new LHApiException(Status.FAILED_PRECONDITION, "UserTaskRun is in status " + status);
+        }
+
+        this.results = req.getResults();
```

Comment:
So, this will overwrite the entire current progress.

Reply:
Yes, that is true. We should document that. It's not a `PATCH` but more of a `POST` I think